src/circuits/simple_gadgets.hpp

```diff
@@ -0,0 +1,110 @@
+#ifndef __ZETH_SIMPLE_GADGETS_HPP__
```

Comment:
`simple_gadgets.hpp` is quite general and does not very well inform on the content of the document. Please rename accordingly (maybe something like `xor_gadgets.hpp` or `blake2s_(components|operations)`?)

Reply:
I wanted to keep a name independent of blake in case we were to reuse it. xor_gadgets would be great if not for the "double_bit32_sum_eq_gadget".

What about something like binary_operation_gadget or herlper_gadgets ?

Reply:
` binary_operation_gadgets` sounds more indicative

Reply:
ok !